src/js/models/layerstructure.js

```diff
@@ -203,6 +203,14 @@ define(function (require, exports, module) {
                     return layer.kind === layer.layerKinds.GROUPEND;
                 });
         },
+        
+        /**
+        * All non-endgroup layers, in reverse order
+        * @type {Immutable.List.<Layer>}
+        */
+        "allVisibleReversed": function () {
+            return this.allVisible.reverse();
```

Comment:
Pretty sure `O(n)` based on my testing and (not particularly useful) reading 'round the Internet